### PR TITLE
[Arctic][AMS] admin-configured default table properties with adding into catalog #171

### LINKED ISSUE #171: [Improvement]Support admin-configured default table properties
Currently, to set user-configued properties of a table, it can be achieved by 1. adding table properties in create table DDL or 2.  add dynamic table properties in a DML. Whereas for a multi-user platform, it is common to share options such as `write.target-file-size-bytes` within a group of users. Thus, administrators should be capable to set service-wide options with default vaules instead of setting shared options by users in their SQLs repeatly.

### PR DESCRIPTION
**Pull request by nicochen**
## Why are the changes needed?
Resolve #171

## Brief change log
  - *Add logical to filter  default properties for a catalog with specific prefix in `TablePropertiesUtil`.*
  - *Add logical to  set default properties into table metadata when creating tablemeta in `TableMetaBuilder`*

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (yes)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

File: core/src/main/java/com/netease/arctic/catalog/BaseArcticCatalog.java [hunk 1]
```diff
@@ -547,6 +548,14 @@ protected ConvertStructUtil.TableMetaBuilder createTableMataBuilder() {
       ConvertStructUtil.TableMetaBuilder builder = ConvertStructUtil.newTableMetaBuilder(
           this.identifier, this.schema);
       String tableLocation = getTableLocationForCreate();
+      try {
+        Map<String, String> catalogProperties = client.getCatalog(this.identifier.getCatalog()).getCatalogProperties();
```
Comment (zhoujinsong) [inline core/src/main/java/com/netease/arctic/catalog/BaseArcticCatalog.java:552]: Can we reuse the properties in  `catalogMeta` ?

File: core/src/main/java/com/netease/arctic/utils/TablePropertyUtil.java [hunk 2]
```diff
@@ -151,4 +153,41 @@ public static long getTableWatermark(Map<String, String> properties) {
...
+  /**
+   * Merge default properties into table properties.
+   * Filter out unnecessary properties from catalog. once 'log-store-enable' or
+   * 'optimize-enable' flag is off, filter out related default properties.
+   *
+   * @param target user-defined table properties
+   * @param catalogProperties catalog properties
+   * @return merged properties
+   */
+  public static Map<String, String> mergeTableProperties(Map<String, String> target,
```
Comment (zhoujinsong) [inline core/src/main/java/com/netease/arctic/utils/TablePropertyUtil.java:166]: Does `CatalogUtil#mergeCatalogPropertiesToTable` method match your requirement yet?

**Review by zhoujinsong — COMMENTED** [review]
@nicochen  Thanks a lot for your work! I left some comments, please take another look.

**Comment by nicochen** [comment]
@zhoujinsong As we talked before, merge filter prefix code into `CatalogUtil` to merge properties  and  call it in `BaseArcticCatalog`

File: core/src/main/java/com/netease/arctic/utils/CatalogUtil.java [hunk 3]
```diff
@@ -185,12 +186,25 @@ public static boolean isIcebergCatalog(ArcticCatalog arcticCatalog) {
...
+    Map<String, String> mergedProperties = catalogProperties.entrySet().stream()
+            .filter(e -> e.getKey().startsWith(CatalogMetaProperties.TABLE_PROPERTIES_PREFIX))
+            .collect(Collectors.toMap(e ->
+                    e.getKey().substring(CatalogMetaProperties.TABLE_PROPERTIES_PREFIX.length()), Map.Entry::getValue));
+
+    if (!PropertyUtil.propertyAsBoolean(tableProperties, TableProperties.ENABLE_LOG_STORE,
+            TableProperties.ENABLE_LOG_STORE_DEFAULT)) {
+      mergedProperties = mergedProperties.entrySet().stream()
+              .filter(e -> !e.getKey().startsWith(CatalogMetaProperties.LOG_STORE_PROPERTIES_PREFIX) &&
+                      !e.getKey().equals("log.consistency.guarantee.enable"))
```
Comment (zhoujinsong) [inline core/src/main/java/com/netease/arctic/utils/CatalogUtil.java:198]: Can we rename the property name `log.consistency.guarantee.enable` to `log-store.consistency-guarantee.enable`?  @YesOrNo828  @zstraw 
We can do this in another PR

File: core/src/main/java/com/netease/arctic/utils/CatalogUtil.java [hunk 4]
```diff
@@ -185,12 +186,25 @@ public static boolean isIcebergCatalog(ArcticCatalog arcticCatalog) {
...
+              .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+    if (!Boolean.parseBoolean(PropertyUtil
```
Comment (zhoujinsong) [inline core/src/main/java/com/netease/arctic/utils/CatalogUtil.java:201]: @nicochen As I know, we should not enable Logstore in catalog default property , but we may enable or disable optimize in catalog default property, so we may consider the default value in catalog properties too ?

**Review by zhoujinsong — COMMENTED** [review]
@nicochen please take another look !

File: core/src/main/java/com/netease/arctic/utils/CatalogUtil.java (hunk 3 above)
Comment (zhoujinsong) [reply core/src/main/java/com/netease/arctic/utils/CatalogUtil.java:198]: #745 complete this.

**Review by zhoujinsong — APPROVED** [review]
LGTM.
